specs/eventing/data-plane.md

```diff
@@ -1,125 +1,167 @@
+1](https://github.com/cloudevents/spec/blob/v1.0.1/http-protocol-binding.md) and
+[HTTP 1.1 protocol](https://tools.ietf.org/html/rfc7230) standards should be
+followed (with the CloudEvents bindings preferred in the case of conflict).
 
-A Sink MAY support `Batched Content Mode` but that mode is not used in Knative
-Eventing currently (that may change in future).
+The current version of this document does not describe protocol negotiation or
+the ability to upgrade an HTTP 1.1 event delivery into a more efficient protocol
+such as GRPC, AMQP or the like. It is expected that a future compatible version
+of this specification might describe a protocol negotiation mechanism.
 
-### Retries
+## Event Delivery
 
-Sinks should expect that retries and accept possibility that duplicate events
-may be delivered.
+To provide simpler support for event sources which might be translating events
+from existing systems, some data plane requirements for senders are relaxed in
+the general case. In the case of Knative Eventing provided resources (Channels
+and Brokers) which implement these roles, requirements are increased from SHOULD
+to MUST. Exceptions with stronger requirements for Channels and Brokers are
+called out as they occur.
 
-### Error handling
+### Minimum supported protocol
 
-If Sink is not returning HTTP success header (200 or 202) then the event may be
-sent again. If the event can not be delivered then some sources of events (such
-as Knative sources, brokers or channels) MAY support
-[dead letter sink or channel](https://github.com/knative/eventing/blob/main/docs/delivery/README.md) for events that can not be
-delivered.
+All senders and recipients MUST support the CloudEvents 1.0 protocol and the
+[binary](https://github.com/cloudevents/spec/blob/v1.0.1/http-protocol-binding.md#31-binary-content-mode)
+and
+[structured](https://github.com/cloudevents/spec/blob/v1.0.1/http-protocol-binding.md#32-structured-content-mode)
+content modes of the CloudEvents HTTP binding. Senders MUST support both
+cleartext (`http`) and TLS (`https`) URLs as event delivery destinations.
+
+### HTTP Verbs
+
+In the absence of specific delivery preferences, the sender MUST initiate
+delivery of the event to the recipient using the HTTP POST verb, using either
+the structured or binary encoding of the event (sender's choice). This delivery
+SHOULD be performed using the CloudEvents HTTP Binding, version 1.0.
+
+Senders MAY probe the recipient with an [HTTP OPTIONS
+request](https://tools.ietf.org/html/rfc7231#section-4.3.7); if implemented, the
+recipent MUST indicate support the POST verb using the [`Allow`
+header](https://tools.ietf.org/html/rfc7231#section-7.4.1). Senders which
+receive an error SHOULD proceed using the HTTP POST mechanism.
+
+### Event Acknowledgement and Repeat Delivery
+
+Event recipients MUST use the HTTP response code to indicate acceptance of an
+event. The recipient MUST NOT return a response accepting the event until it has
```

Comment:
There is no way to check this behavior (which I find too limiting anyway, see my comment above about QoS). I would either remove this sentence or make it non-normative.

Reply:
I switched this to SHOULD; I agree that this may be difficult to test for conformance, so I could be convinced to drop it, but it also feels like useful guidance.